libs/permissions/permission-manager-impl/src/main/kotlin/net/corda/libs/permissions/manager/impl/RbacBasicAuthenticationService.kt

```diff
@@ -28,22 +39,22 @@ class RbacBasicAuthenticationService(
         val clearTextPassword = String(password)
 
         if (repeatedLogonsCache.verifies(loginName, clearTextPassword)) {
-            return true
+            return AuthenticationState(true, null)
         }
 
-        val user = permissionManagementCache.getUser(loginName) ?: return false
+        val user = permissionManagementCache.getUser(loginName) ?: return failedAuthentication
 
         if (user.saltValue == null || user.hashedPassword == null) {
-            return false
+            return failedAuthentication
         }
 
         if (passwordService.verifies(clearTextPassword, PasswordHash(user.saltValue, user.hashedPassword))) {
             repeatedLogonsCache.add(loginName, clearTextPassword)
-            return true
+            return AuthenticationState(true, checkExpiryStatus(user.passwordExpiry))
```

Comment:
Having thought a bit more about it, I thought it would be a good idea not to add the item of the cache if password expired or close to expiry. As it may be stuck in cache for sometime and confuse repeated logon attempts. Therefore proposing:
```suggestion
            val expiryStatus = checkExpiryStatus(user.passwordExpiry)
            // Ensure that only active status if cached
            if (expiryStatus == ExpiryStatus.ACTIVE) {
                repeatedLogonsCache.add(loginName, clearTextPassword)
            } else {
                repeatedLogonsCache.remove(loginName)
            }
            return AuthenticationState(true, expiryStatus)
```